velox/functions/prestosql/tests/CastBaseTest.h

```diff
@@ -249,6 +249,91 @@ class CastBaseTest : public FunctionBaseTest {
             makeRowVector({makeNullableFlatVector<TFrom>(input, fromType)})),
         expectedErrorMessage);
   }
+
+  void testComplexCast(
+      const std::string& fromExpression,
+      const VectorPtr& data,
+      const VectorPtr& expected,
+      bool nullOnFailure = false) {
```

Comment:
The existing method `testCast(const TypePtr& fromType, const TypePtr& toType, const VectorPtr& input, const VectorPtr& expected)` should also be able to test complex-typed castings. I saw there are some differences, such as the test of input vector doesn't change after the evaluation. Might it be possible to combine these two methods into one to reduce redundant interfaces?

Reply:
Makes sense. I assume it's better to do that in a separate PR. Opened https://github.com/facebookincubator/velox/pull/8254 for this change. @kagamiori @mbasmanova Could you help review? Thanks.